tests/test_dataset.py

```diff
@@ -0,0 +1,29 @@
+import shutil
+
+import msprime
+import pytest
+import xarray as xr
+
+import phylokit as pk
+
+
+class TestCreateTreeDataset:
+    def create_test_dataset(self):
+        tsa = msprime.sim_ancestry(
+            samples=100, sequence_length=100, ploidy=1, random_seed=1234
+        )
+        ts = msprime.sim_mutations(tsa, rate=0.03, random_seed=1234)
+        ds = pk.from_tskit(ts.first())
+        return ds
+
+    @pytest.fixture
```

Comment:
This seems a bit overly elaborate, how about:
```python
def example_datasets():
    # return a list of example datasets containing difference mixtures of stuff we'll have in phylokit

@pytest.mark.parametrize("ds", example_datasets())
def test_file_roundtrip(ds, tmp_path):  
          path = tmp / "test.zarr"
          pk.save_dataset(ds,  path)
          ds2 = pk.open_dataset(path)
          xr.testing.assert_identical(ds, ds2)
```

should also test when we pass in a string (but doesn't need to go over different example datasets.